### ISSUE TITLE
While minifying app stylesheet: path.js:8:11: Path must be a string. 

### ISSUE THREAD

**Issue by streamnsight** [meteor/meteor#7523]
While minifying app stylesheet:
path.js:8:11: Path must be a string. Received null
at assertPath (path.js:8:11)
at Object.posix.relative (path.js:496:3)
at pathRelative (packages/minifier-css/minifier.js:206:1)
at packages/minifier-css/minifier.js:165:1
at Array.forEach (native)
at Function..each..forEach (packages/underscore/underscore.js:105:1)
at packages/minifier-css/minifier.js:122:1
at Array.forEach (native)
at Function..each..forEach (packages/underscore/underscore.js:105:1)
at rewriteRules (packages/minifier-css/minifier.js:99:1)
at Object.CssTools.rewriteCssUrls (packages/minifier-css/minifier.js:85:1)
at packages/minifier-css/minifier.js:55:1
at Array.forEach (native)
at Function..each..forEach (packages/underscore/underscore.js:105:1)
at Object.CssTools.mergeCssAsts (packages/minifier-css/minifier.js:32:1)
at packages/minifyStdCSS/plugin/minify-css.js:79:1
at CssToolsMinifier.processFilesForBundle
(packages/minifyStdCSS/plugin/minify-css.js:17:1)

seems to be linked to gulp, according to 
http://stackoverflow.com/questions/28140012/gulp-typeerror-arguments-to-path-join-must-be-strings

which means it could be any package...
How do i sort this out?

**Comment by darraghmckay** [comment]
Did you get this right after updating to v1.4? 
I've the exact same issue

**Comment by Exilz** [comment]
I'm using `fourseven:scss`, so I figured out this issue is probably related to it.

I tried to remove the default minifier and use the one fourseven is recommending.

```
meteor remove standard-minifier-css
meteor add seba:minifiers-autoprefixer
```

This worked for me, altough I don't feel comfortable replacing the "official" one.

**Comment by karolselak** [comment]
I had the same issue and yes, it is connected with css paths. I use paths in connection with SVG filters, so when I comment it, all shadows in my style disappear. I can bypass it by setting style inline in HTML (or JSX in my case), but that's very ugly solution. After all, it works, so we can use this bypass until the issue is solved by meteor team.

**Comment by abernix** [comment]
This sounds identical to #7446 which I just closed – you should review it for similarities.  Even if you're not using the same package, the likely answer is that there has been a package update that you haven't done, or a package that you're using has been deprecated.   Please make sure you're the latest versions of packages which performing minifier actions as the API has changed.  Including the output of `meteor list` would probably be helpful, but if you systematically remove packages from your project until the error goes away, you'll find the problem.

**Comment by karolselak** [comment]
@abernix, you are not right, just take a look at my previous comment. Please reopen this issue, because it occures also in pure meteor project. Just create new app and paste
`body {
  filter: url(#xyz);
}`
into main.css file - you'll get the same error.

**Comment by abernix** [comment]
@karolselak First of all, how do you know your issue is the same one as the original poster?  The error message is generic.  #7394 is also similar.

If you feel your usage is in-fact a bug, open a new issue with a reproduction of the exact problem you are referring to since there is no way of knowing what the original poster is having issues with.  I'll consider reopening with a reproduction from the original author of this issue if they confirm that are having the exact same problem for the same reason.

But, I will warn, the syntax that you are using doesn't seem to be covered in [the Mozilla spec for the `filter` attribute](https://developer.mozilla.org/en/docs/Web/CSS/filter).  It explicitly states that:

> The url() function takes the location of an XML file that specifies an SVG filter, and may include an anchor to a specific filter element.

The error goes away if I change use that definition.  If you're going to open a new issue, please make sure to include a reference to a relative spec that explains how your usage is expected to work.

**Comment by TimHeckel** [comment]
For what it's worth, to at least pinpoint where the potential css problem lies, I was able to `console.log` in the below file in the `rewriteRules` function, inside the _.each(rule.declarations` loop (I simply did a `console.log("declaration is ", declaration);` and the offending line was spit out when running `meteor` (1.4.1.1)

`~/.meteor/packages/standard-minifier-css/.1.2.0.1agfg4z++os+web.browser+web.cordova/plugin.minifyStdCSS.os/packages/minifier-css.js`

**Comment by llvasconcellos** [comment]
Same problem with me. Meteor 1.4.
Removing the bellow line fixed the problem. It came with qtip tooltip plugin styles. 
```css
behavior: url(#default#VML);
```

**Comment by abernix** [comment]
Just an FYI, this is probably fixed by [this line](https://github.com/meteor/meteor/blob/705a0c17095c84086c35371b41729a38a8600078/packages/minifier-css/minifier.js#L141) in #8048 – which will most likely surface with/around Meteor 1.4.3.

**Comment by glasser** [comment]
@abernix Is it possible to publish a new version of the package so we can try it out before 1.4.3?  (I'd do it myself but I am pretty rusty with this repo!)

**Comment by abernix** [comment]
@glasser Yes, it should be possible, and I think it could be done in a way to avoid the problem [you encountered with `coffeescript`](https://github.com/meteor/meteor/pull/7653#issuecomment-267018164).  Although, since Meteor 1.4.3-beta.1 was released with this change already, this could also be tested right now by updating to that beta:

    meteor update --release 1.4.3-beta.1

**Comment by glasser** [comment]
Great, this seems to work.

If I want to make as minimal a change as possible, adding this to `.meteor/packages` appears to work:

```
# Fix https://github.com/meteor/meteor/issues/7523. Note that even though the
# change is in minifier-css, the plugin is pre-built inside
# standard-minifier-css so that's the one that really needs to be updated.
standard-minifier-css@1.3.3-beta.1
minifier-css@1.2.16-beta.1
```

**Comment by jeanfredrik** [comment]
The solution proposed by @glasser works well for me in 1.4.2.3, but not in 1.4.3.1. Steps to reproduce:

```
meteor create example --release 1.4.3.1
cd example
meteor npm install
meteor
```

Add this to client/main.css:

```css
.example {
  filter: url('#foo-bar');
}
```

You will get the following error:

```
=> Errors prevented startup:                  
   
   While minifying app stylesheet:
   path.js:8:11: Path must be a string. Received null
   at assertPath (path.js:8:11)
   at Object.posix.relative (path.js:496:3)
   at pathRelative (packages/minifier-css.js:449:30)
   at packages/minifier-css.js:408:31
   at Array.forEach (native)
   at Function._.each._.forEach (packages/underscore.js:139:11)
   at packages/minifier-css.js:365:7
   at Array.forEach (native)
   at Function._.each._.forEach (packages/underscore.js:139:11)
   at rewriteRules (packages/minifier-css.js:342:5)
   at Object.CssTools.rewriteCssUrls (packages/minifier-css.js:328:5)
   at packages/minifier-css.js:298:16
   at Array.forEach (native)
   at Function._.each._.forEach (packages/underscore.js:139:11)
   at Object.CssTools.mergeCssAsts (packages/minifier-css.js:275:7)
   at packages/minifyStdCSS_plugin.js:113:31
   at CssToolsMinifier.processFilesForBundle (packages/minifyStdCSS_plugin.js:33:16)
```

**Comment by abernix** [comment]
@jeanfredrik Thanks for pointing this out.  I'm trying to fix this right now with #8437.  Stay tuned to that PR!
